src/library/features/crates/cratemanager.h

```diff
@@ -0,0 +1,101 @@
+#ifndef MIXXX_CRATEMANAGER_H
+#define MIXXX_CRATEMANAGER_H
+
+#include <QSet>
+#include <QList>
+
+#include "library/crate/crate.h"
+#include "track/trackid.h"
+#include "util/db/sqlstorage.h"
+
+#include "library/crate/cratesummary.h"
+#include "library/features/crates/cratestoragehelpers.h"
+#include "library/features/crates/cratestorage.h"
+#include "library/features/crates/cratetracks.h"
+#include "library/features/crates/cratehierarchy.h"
+
+#include "util/db/fwdsqlqueryselectresult.h"
+#include "util/db/sqlsubselectmode.h"
+
+class CrateManager : public QObject, public virtual SqlStorage {
+    Q_OBJECT
+
+  public:
+    CrateManager() = default;
+    ~CrateManager() override = default;
+
+    void repairDatabase(
+      QSqlDatabase database) override;
+
+    void connectDatabase(
+            QSqlDatabase database) override;
+    void disconnectDatabase() override;
+
+    ///////////////////
+    // Member access //
+    ///////////////////
+
+    const inline CrateStorage& storage() const {
+        return m_crateStorage;
+    }
+
+    const inline CrateTracks& tracks() const {
+        return m_crateTracks;
+    }
+
+    const inline CrateHierarchy& hierarchy() const {
+        return m_crateHierarchy;
+    }
+
+    //////////////////////////////////////
+    // wrappers for non const fucntions //
+    // to be called by trackCollection  //
+    //////////////////////////////////////
+
+    bool onPurgingTracks(const QList<TrackId>& trackIds);
+
+    ////////////////////////////////////////////////////////
+
+    void crateCrate();
+    void deleteCrate();
+    void moveCrate();
+    void renameCrate();
+    void updateCrateTracks();
+    void checkClosure() const;
+
+    ////////////////////////////////////////////////////////
+
+    bool insertCrate(const Crate& crate, CrateId* pCrateId = nullptr);
+    bool updateCrate(const Crate& crate);
```

Comment:
Updating a crate means changing it's name, autoDj or locked status. I find it a fine name for the function. I'll add a comment